### PR TITLE
Allure steps for s3-tests. Fixes #39

### LINKED ISSUE #39: Test reports missing a lot of data
## Current Behavior

https://http.fs.neo.org/7ZVSQHtioVRiDT2btN9xEmQhczXV9vryCEAHUawfD8Sc/7-/index.html#suites/6d7c9ec7c4f3e4460ae14fd266a64e04/bf44f995406cbfbf/

Frankly, I don't understand a thing from it. There is just one ` nose_setup 150ms ` step that has zero logs.

## Expected Behavior
Proper steps, console outputs, logs.

### PR DESCRIPTION
**Pull request by MaxGelbakhiani**
Followed @evgeniiz321 review advice and implemented client methods overload with allure wraps.

### CODE REVIEW

File: s3tests_boto3/functional/__init__.py [hunk 1]
```diff
@@ -14,18 +15,23 @@
...
+from functools import wraps
 
 config = munch.Munch
 
 # this will be assigned by setup()
 prefix = None
 
+boto3._get_default_session = lambda: WrappedSession()
 
+@allure.step
```
Comment (roman-khimov) [inline s3tests_boto3/functional/__init__.py:28]: Nameless? What's it gonna look like in the log?

**Comment by roman-khimov** [comment]
What's up with this? No updates in two days.

**Comment by MaxGelbakhiani** [comment]
> What's up with this? No updates in two days.

Added names to allure steps and decorator for functions getting settings from config. Also removed @allure.step from some functions where necessary in compare to the original commit.

File: s3tests_boto3/functional/test_s3_neofs.py [hunk 2]
```diff
@@ -27,6 +28,8 @@
 from .utils import _get_status, _get_status_and_error_code, assert_raises
 
 
+
+@allure.step
```
Comment (roman-khimov) [inline s3tests_boto3/functional/test_s3_neofs.py:32]: Left intentionally (and others in the same file)?

Comment (MaxGelbakhiani) [reply s3tests_boto3/functional/test_s3_neofs.py]: Fixed

File: s3tests_boto3/functional/utils.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
+import allure
 import random
 import string
 
 
+@allure.step("Assert Raises")
```
Comment (roman-khimov) [inline s3tests_boto3/functional/utils.py:6]: Is it a proper step, btw?

**Review by roman-khimov — APPROVED** [review]

File: s3tests_boto3/functional/utils.py (hunk 3 above)
Comment (MaxGelbakhiani) [reply s3tests_boto3/functional/utils.py:6]: In my opinion, it is. As for tests, checking negative scenarios assertion with expected error is just a step. Like in [test_object_copy_bucket_not_found](https://github.com/nspcc-dev/s3-tests/blob/a35437900562ae3b17797fe82ca7c2c1840665fe/s3tests_boto3/functional/test_s3.py#L6768C56-L6768C56)

And it looks like this in test report:
image

**Pull request merged**